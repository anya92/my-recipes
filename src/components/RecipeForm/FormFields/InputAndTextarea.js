import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Label from '../styled/Label';
import StyledInput from '../styled/Input';
import StyledTextarea from '../styled/Textarea';

const FormField = styled.div`
  div {
    display: flex;
    span {
      font-size: 1.1rem;
      background-color: #F2F2F2;
      display: inline-flex;
      align-items: center;
      padding: 0 5px;
      border-left: 2px solid #333;
    }
  }
`;

export const Input = ({
  name, value, handleInputChange, additionalText, type = 'text', error = false, isRequired,
}) => (
  <FormField>
    <Label htmlFor={name}>{name} {isRequired && '*'}</Label>
    <div>
      <StyledInput
        type={type}
        min={type === 'number' ? 0 : ''}
        id={name}
        value={value}
        onChange={e => handleInputChange(e, name)}
        error={error}
        additionalText={additionalText}
      />
      {additionalText && <span>{additionalText}</span>}
    </div>
  </FormField>
);

Input.defaultProps = {
  additionalText: '',
  type: 'text',
  error: false,
  isRequired: false,
};

Input.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  handleInputChange: PropTypes.func.isRequired,
  additionalText: PropTypes.string,
  type: PropTypes.string,
  error: PropTypes.bool,
  isRequired: PropTypes.bool,
};

export const Textarea = ({
  name, value, handleInputChange, error, isRequired,
}) => (
  <FormField>
    <Label htmlFor={name}>{name} {isRequired && '*'}</Label>
    <StyledTextarea
      id={name}
      rows={4}
      value={value}
      onChange={e => handleInputChange(e, name)}
      error={error}
    />
  </FormField>
);

Textarea.defaultProps = {
  error: false,
  isRequired: false,
};

Textarea.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  error: PropTypes.bool,
  isRequired: PropTypes.bool,
};
