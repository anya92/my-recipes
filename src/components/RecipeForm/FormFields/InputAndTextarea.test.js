import React from 'react';
import { shallow } from 'enzyme';

import { Input, Textarea } from './InputAndTextarea';

let name;
let value;
let handleInputChange;
let wrapper;

describe('Input', () => {
  beforeEach(() => {
    name = 'name';
    value = '';
    handleInputChange = jest.fn();
    wrapper = shallow(<Input
      name={name}
      value={value}
      handleInputChange={handleInputChange}
    />);
  });

  test('should render Input correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should call handleInputChange on input change', () => {
    const inputValue = 'Spaghetti';
    wrapper.find('Input').simulate('change', { target: { value: inputValue } });
    expect(handleInputChange).toHaveBeenLastCalledWith({ target: { value: inputValue } }, name);
  });
});

describe('Textarea', () => {
  beforeEach(() => {
    name = 'directions';
    value = '';
    handleInputChange = jest.fn();
    wrapper = shallow(<Textarea
      name={name}
      value={value}
      handleInputChange={handleInputChange}
    />);
  });

  test('should render Textarea correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should call handleInputChange on textarea change', () => {
    const textareaValue = 'New directions.';
    wrapper.find('Textarea').simulate('change', { target: { value: textareaValue } });
    expect(handleInputChange).toHaveBeenLastCalledWith({ target: { value: textareaValue } }, name);
  });
});
