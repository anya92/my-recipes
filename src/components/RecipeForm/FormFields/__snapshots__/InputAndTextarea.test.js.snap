// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input correctly 1`] = `
<styled.div>
  <styled.label
    htmlFor="name"
  >
    name
     
  </styled.label>
  <div>
    <Input
      additionalText=""
      error={false}
      id="name"
      min=""
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
</styled.div>
`;

exports[`Textarea should render Textarea correctly 1`] = `
<styled.div>
  <styled.label
    htmlFor="directions"
  >
    directions
     
  </styled.label>
  <Textarea
    error={false}
    id="directions"
    onChange={[Function]}
    rows={4}
    value=""
  />
</styled.div>
`;
